src/Http/Http.Abstractions/src/Routing/EndpointMetadataCollection.cs

```diff
@@ -215,4 +218,27 @@ public void Reset()
             _current = null;
         }
     }
+
+    private sealed class EndpointMetadataCollectionDebugView
+    {
+        private readonly EndpointMetadataCollection _collection;
+
+        public EndpointMetadataCollectionDebugView(EndpointMetadataCollection collection)
+        {
+            ArgumentNullException.ThrowIfNull(collection);
+
+            _collection = collection;
+        }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+        public object[] Items
+        {
+            get
+            {
+                var items = new object[_collection.Count];
```

Comment:
I haven't made a debugger type proxy in a while - we're returning a copy to protect it from mutation in the locals/watch window?

Reply:
I don't know exactly. All the examples I saw in dotnet/runtime copied the collection, so I followed them.